paasta_tools/frameworks/native_scheduler.py

```diff
@@ -400,6 +450,10 @@ def recreate_drain_method(self):
             **self.service_config.get_drain_method_params(self.service_config.service_namespace_config)
         )
 
+    def reload_constraints(self):
+        if len(self.constraints) == 0 and self.service_config.get_constraints():
+            self.constraints = self.service_config.get_constraints()
```

Comment:
Why copy the config's constraints into self?

Also, won't the `len(self.constraints) == 0` conditional prevent us from ever changing constraints?